internal/controller/datadogagent/controller_reconcile_agent.go

```diff
@@ -297,6 +297,11 @@ func (r *Reconciler) labelNodesWithProfiles(ctx context.Context, profilesByNode
 	for nodeName, profileNamespacedName := range profilesByNode {
 		isDefaultProfile := agentprofile.IsDefaultProfile(profileNamespacedName.Namespace, profileNamespacedName.Name)
 
+		// in the refactor, we don't explicitly set the default profile so empty profile nodes fall under the default profile
+		if profileNamespacedName.Name == "" && profileNamespacedName.Namespace == "" {
+			isDefaultProfile = true
+		}
```

Comment:
Isn't this same as what we already have on line 298?
https://github.com/DataDog/datadog-operator/blob/c95695429c6fa8786bef75501f8e8671e5fa73cc/pkg/agentprofile/agent_profile.go#L197-L198